File: server/Models/Krevet.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace server.Models
{
    [Table("Kreveti")]
    public class Krevet
    {
        [Key]
        [Column("ID")]
        public int ID { get; set; }
        
        [Column("Pacijent")]
        public Pacijent pacijent { get; set; }

        [Column("Zauzet")]
        public bool zauzet { get; set; }

        [JsonIgnore]
        public Soba soba { get; set; }
    }

}

File: client/krevet.js
import {Pacijent} from "./pacijent.js"

export class Krevet{

    constructor(id)
    {
        this.id = id;
        this.zauzet = false;
        this.pacijent = null;
    }

    oslobodiKrevet() 
    {
        this.pacijent = null;
        this.zauzet = false;
    }

    smestiUKrevet(pacijent)
    {
        this.pacijent = pacijent;
        this.zauzet = true;
    }

    slobodan(){
        return !this.zauzet;
    }
    
}